docpipeline/generate.py

```diff
@@ -91,6 +92,7 @@ def format_docfx_json(metadata):
 def process_blob(blob, credentials, devsite_template):
     tmp_path = pathlib.Path(tempfile.TemporaryDirectory(prefix="doc-pipeline.").name)
     api_path = tmp_path.joinpath("obj/api")
+    decompress_path = api_path
```

Comment:
is there a reason to not remove `api_path` as a name and replace it with `decompress_path`? I can see how that is a more understand name, and I see calls to `api_path` have been updated as well.

Reply:
`decompress_path` only gets used to determine where we should decompress the tar content into, whereas `api_path` gets used a lot more further down. I have tried just using `api_path` alone but when we have a condition where `api/*` directory is encountered in the tarball, then we'd have to update it before decompressing (from `/obj/api` to `obj`), then update it back to `obj/api` which I thought was a bit too much unnecessary changes.

Ultimately it needs to end up in `obj/api` so I only wanted to use and update a separate variable `decompress_path`.

Reply:
The bit I am 'clicking into' is that this is aliasing another variable, within the same function. My take on this is that either
1) this is unnecessary and doesn't add clarity
2) this is a new more informative name
3) this is a sign this area should be abstracted.

At minimum I would change this to be api_path=decompress_path=value